lib/teach-api.js

```diff
@@ -111,6 +111,17 @@ _.extend(TeachAPI.prototype, {
                      url + '. Not sending auth token.');
       }
     }
+ 
+    // do we need credentials + Django CSRF token?
+    if (['post','put','delete'].indexOf(method) > -1) {
+      var csrf = document.cookie.match(/csrftoken=\S+/);
+      if (csrf) {
+        csrf = csrf[0].replace(';','').split('=');
+        var token = csrf[1];
+        req.set('X-CSRFtoken', token ? token : "FALSE");
```

Comment:
I went on the example code in https://docs.djangoproject.com/en/1.9/ref/csrf, which gives:
```
function csrfSafeMethod(method) {
    // these HTTP methods do not require CSRF protection
    return (/^(GET|HEAD|OPTIONS|TRACE)$/.test(method));
}
$.ajaxSetup({
    beforeSend: function(xhr, settings) {
        if (!csrfSafeMethod(settings.type) && !this.crossDomain) {
            xhr.setRequestHeader("X-CSRFToken", csrftoken);
        }
    }
});
```
And this header seems to work, when the token is correct.